recipe/patches/0006-BLD-Link-to-cudart-statically.patch

```diff
@@ -0,0 +1,55 @@
+From 203a699d5e186468b4aa396eae1a194e81e29f23 Mon Sep 17 00:00:00 2001
+From: Daniel Ching <dching@nvidia.com>
+Date: Mon, 28 Apr 2025 18:57:54 -0500
+Subject: [PATCH 6/6] BLD: Link to cudart statically
+
+---
+ cmake/Dependencies.aarch64-linux.cmake | 3 ++-
+ cmake/Dependencies.cmake               | 3 ++-
+ cmake/modules/FindNVJPEG.cmake         | 2 +-
+ 3 files changed, 5 insertions(+), 3 deletions(-)
+
+diff --git a/cmake/Dependencies.aarch64-linux.cmake b/cmake/Dependencies.aarch64-linux.cmake
+index ac010892..cae40eda 100644
+--- a/cmake/Dependencies.aarch64-linux.cmake
```

Comment:
It not the file you are looking for.
`Dependencies.aarch64-linux.cmake` is used only for Jetson which is identified as `aarch64-linux` https://github.com/NVIDIA/DALI/blob/main/docker/Dockerfile.build.aarch64-linux#L125. SBSA (just aarchd64) doesn't use it.

Reply:
It's not a duplicate. It reverts part of the first patch (replace all static links with dynamic links (non-dlopen)). Since the goal is to make CUDA optional, then we must revert to statically linking to cudart because there is no dlopen wrapper for cudart (or it is unused in some places?).

Reply:
> then we must revert to statically linking to cudart because there is no dlopen wrapper for cudart (or it is unused in some places?).
That is deliberate. We want ot utilize the minor version compatibility -compile with the latest CTK and run with any. It works as long as there is no missing symbols. In case of libs their API is stable, but runtime provides more features over time and we don't want to create fallbacks on our end.
The runtime is small so it is not a big deal to ship linked with DALI.